### ISSUE TITLE
Benchmark Inference Times - Cerebros Model vs. Original GPT-2

### ISSUE THREAD

**Issue by david-thrower** [david-thrower/cerebros-core-algorithm-alpha#154]
# Add a test that better highlights our competitive advantage:

## Objective

To demonstrate the ON timing advantage of the Cerebros neuromorphic model, we need a clear side-by-side comparison of inference times (cold start and warm) between:

- Original GPT-2 text classifier Model: Uses the full transformer architecture.
- Cerebros Model: Replaces GPT-2's transformer blocks with neuromorphic components, but still uses the same tokenizer and preprocessing. (The current model being tested)...

### Re - include the Original GPT-2 Model baseline which was removed for efficiency purposes: 

- [x] Basically create a second GPT subclass that **includes** the GPT backbone in __call__() and sets encoder layers as trainable.
- [x] Then add a second Model() suing a simple binary logistic output Dense(1, "sigmoid") without the Cerebros layer augmentation.
- [x] Fine tune the standard GPT model on the same data set we are using.

## Add a timing comparrison:

- [x] Add timing to this and the Cerebros model and print out the timing difference.

## Scale down data set:

- [ ] Modify the subset of the data if necessary to enable both model trainings to commence in a timing reasonable given the resource limits with Github workflows. The current task takes over an hour to complete, and the transformer model will take even longer.

## Caveats: Code needs manually written. All AI tools asked to suggest the code changes seem to miss major details to the point it will take longer to fix than to just write it manually. This is likely an artifact of less commonly used components that need to be replaced soon when we have time (e. g. KerasNLP that needs replaced with huggingface transformers). We have tried all of these, but are getting back code suggestions that are structurally wrong:

- DeepSeek-R1-Distill-Qwen-32B
- R1 full 600B 
- QwQ-32B

**Comment by david-thrower** [comment]
Ok, so we set up a field that is slanted in the favor of Transformers at a 90 degree slant and making a side - by - side comparrison.

## What we give them in our comparison:

✅ The GPT-2 model started benefitting from pre-trained weights.
✅ We fine tune the GPT pre-trained model for one hour with all weights trainable. Only a sequence length of 96 will work on the machine that GitHub gives us and that takes an hour just to fine tune for 4 epochs on it.

## What we give ourselves in the comparison:

✅ We are using models around 20M weights.    
✅ We started from a cold start on a blank Glorot / Xavier random slate. 
✅ We are training 5 models for 15 epochs, each from a blank slate, in about the same time as we can train their model for only 4 epochs on the same data. 
✅ **We are running this on a sequence length of 250, which between running 5 times as many models, for almost 4 times as many epochs, on a little under 3 times the sequence length, we show clearly that we scale at ON timing as opposed to the GPT scaling in ON**2 timing with regard to sequence length and other unfavorable ON**2 parameters.**

**Comment by david-thrower** [comment]
# A better comparison is included in #172. This needs to be added to the docs though.
